reframe/core/namespaces.py

```diff
@@ -40,7 +42,7 @@ def __getitem__(self, key):
         return self._namespace[key]
 
     def __setitem__(self, key, value):
-        if key not in self._namespace:
+        if key not in self._namespace and key not in directives.NAMES:
```

Comment:
I don't think this belongs here. I'd vote for specialising this class with this particular behaviour only for the directives namespace.

Reply:
Indeed, that's a leftover!